src/parser/message.rs

```diff
@@ -57,35 +59,40 @@ pub fn message_set<'a>(input: &'a [u8], size: i32) -> IResult<&'a [u8], MessageS
     if size >= 0 {
       take!(i, size as usize)
     } else {
-      Error(Code(ErrorKind::Custom(InputError::InvalidMessageSetSize.to_int())))
+      Error(ErrorKind::Custom(InputError::InvalidMessageSetSize.to_int()))
     }
   };
 
-  flat_map!(input, ms_bytes, |msb| {
-    chain!(
-      msb,
-      messages: message_set_messages,
-      || {
-        messages
-      })
-  })
+  flat_map!(input, ms_bytes, |msb| toto(msb))
+}
+
+fn toto<'a>(msb: &'a [u8]) -> IResult<&'a [u8], MessageSet<'a>> {
+  do_parse!(
```

Comment:
this could be replaced with `messages_set_messages(msb)`. So `flat_map!(input, ms_bytes, |msb| toto(msb))` could be replaced with `flat_map!(input, ms_bytes, messages_set_messages)`

Reply:
done in 7e1ec21